Extension/src/common.ts

```diff
@@ -972,9 +987,9 @@ export function extractCompilerPathAndArgs(inputCompilerPath?: string, inputComp
             // Input has no quotes but can have a compiler path with spaces and args.
             // Go from right to left checking if a valid path is to the left of a space.
             let spaceStart: number = compilerPath.lastIndexOf(" ");
-            if (spaceStart !== -1 && (!isWindows || !compilerPath.endsWith("cl.exe")) && !checkFileExistsSync(compilerPath)) {
+            if (spaceStart !== -1 && (!isWindows || !compilerPath.toLowerCase().endsWith("cl.exe")) && !checkFileExistsSync(compilerPath)) {
                 let potentialCompilerPath: string = compilerPath.substr(0, spaceStart);
-                while ((!isWindows || !potentialCompilerPath.endsWith("cl.exe")) && !checkFileExistsSync(potentialCompilerPath)) {
+                while ((!isWindows || !potentialCompilerPath.toLowerCase().endsWith("cl.exe")) && !checkFileExistsSync(potentialCompilerPath)) {
```

Comment:
Also, this check might match `clang-cl.exe` as well.  I think it needs to grab the base filename from potentialCompilerPath and check the whole name.

Reply:
Or something like compilerPath.toLowerCase().endsWith("\\cl.exe") || compilerPath.toLowerCase().endsWith("/cl.exe") || compilerPath.toLowerCase() === "cl.exe"

Reply:
I updated the function and removed the logic for checking `clang-cl.exe` as that's not needed anymore.